scripts/cray_runall.sh

```diff
@@ -22,14 +58,75 @@ my_exit() {
     exit $tests_failed
 }
 
+#
+# srun specific function to run the client/server tests
+# in simpledirectory.  This function assumes the client and
+# server are run on the same node.
+#
+
+run_client_server_srun() {
+    local prog_name=$1
+    local fabric=$2
+    echo Number of args to run_client_server_srun is $#
+    if [ $# -gt 2 ]; then
+        local server_args=$3
+    fi
+    if [ $# -gt 3 ]; then
+        local client_args=$4
+    fi
+    local gni_ip_addr=`/sbin/ifconfig ipogif0 | grep 'inet addr:' | cut -d: -f2 | awk '{ print $1}'`
+    if [ -f mpmd_conf_file ] ; then
+        rm mpmd_conf_file
+    fi
+
+    if [ -z $server_args ]; then
+        echo  "0  run_client_server.sh -t $prog_name -f $fabric">> mpmd_conf_file
+    else
+        echo server_args=$server_args
+        echo  "0  run_client_server.sh -t $prog_name -f $fabric -a $server_args">> mpmd_conf_file
+    fi
+
+    if [ -z $client_args ]; then
+        echo  "1  run_client_server.sh -t $prog_name -f $fabric -c ">> mpmd_conf_file
+    else
+        echo client_args=$client_args
+        echo  "1  run_client_server.sh -t $prog_name -f $fabric -c -a $client_args">> mpmd_conf_file
+    fi
+
+#   cat mpmd_conf_file
+#
+# the exclusive option seems necessary for MPMD
+#
+    $launcher --multi-prog --exclusive -N 1 -t $timeout mpmd_conf_file
+    return $?
+}
+
+#
+# the aprun method assume one is running the script
+# in a batch environment, so the first invocation of
+# aprun will put a rank on the same node as for the
+# second invocation.
+#
+
+run_client_server_aprun() {
+    local prog_name=$1
+    local gni_ip_addr=`aprun -n 1 /sbin/ifconfig ipogif0 | grep 'inet addr:' | cut -d: -f2 | awk '{ print $1}'`
+    echo  "0: $prog_name" >> mpmd_conf_file
+    echo  "1: $prog_name $gni_ip_addr" >> mpmd_conf_file
+    $launcher -n 1 $prog_name : -n 1 $ prog_name $gni_ip_addr
+    return $?
+}
+
 if [ -x /opt/slurm/default/bin/srun ]; then
```

Comment:
I just realized something.  This path may exist on systems that run both Slurm and PBS-style launchers.  The more accurate thing to do would be to check to see if the module is loaded.  Let me know if you agree, and I'll open an issue to fix this.